blivet/devices/disk.py

```diff
@@ -613,3 +613,49 @@ def dracutSetupArgs(self):
                                             ":".join(opts))])
         else:
             return set(["rd.dasd=%s" % self.busid])
+
+class NVDIMMNamespaceDevice(DiskDevice):
+
+    """ Non-volatile memory namespace """
+    _type = "nvdimm"
+
+    def __init__(self, device, **kwargs):
+        """
+            :param name: the device name (generally a device node's basename)
+            :type name: str
+            :keyword exists: does this device exist?
+            :type exists: bool
+            :keyword size: the device's size
+            :type size: :class:`~.size.Size`
+            :keyword parents: a list of parent devices
+            :type parents: list of :class:`StorageDevice`
+            :keyword format: this device's formatting
+            :type format: :class:`~.formats.DeviceFormat` or a subclass of it
+            :keyword mode: mode of the namespace
+            :type mode: str
+            :keyword devname: name of the namespace (e.g. 'namespace0.0')
```

Comment:
How is `devname` different from `name`? Is the former more like a label?

Reply:
Name is something like `pmem0` (and it is what udev see as name) and devname is `namespace0.0` (this is nvdimm/ndctl "internal" name). devname is a "unique" identifier for the nvdimm "device". For example name depends on the current mode -- after reconfiguring the namespace to sector mode, the name will be changed to `pmem0s`. `namespace0.0` can be used also for nvdimms not visible by udev (without a block device, e.g. in memory mode).